src/components/victory-axis.jsx

```diff
@@ -8,116 +8,251 @@ class VictoryAxis extends React.Component {
+      // this.state.scale
+      domain = this.props.scale().domain();
+    }
+    return this.isVertical() ? domain.reverse() : domain;
+  }
+
+  getRange() {
+    const style = this.getStyles();
+    const margin = this.getMarginValues();
+    if (this.isVertical()) {
+      return [margin.top, parseInt(style.height, 10) - margin.bottom];
+    } else {
+      return [margin.left, parseInt(style.width, 10) - margin.right];
+    }
+  }
+
+  isVertical() {
+    const orientation = this.props.orientation;
+    return (orientation === "left" || orientation === "right");
+  }
+
+  getTransform() {
+    const orientation = this.props.orientation;
+    const margin = this.getMarginValues();
+    const style = this.getStyles();
+    const offset = {
+      top: [0, margin.top],
+      bottom: [0, (parseInt(style.height, 10) - margin.bottom)],
+      left: [margin.left, 0],
+      right: [(parseInt(style.width, 10) - margin.right), 0]
+    };
+    return "translate(" + offset[orientation][0] + "," + offset[orientation][1] + ")";
+  }
+
+  setupScale() {
+    const scale = this.props.scale().copy();
+    scale.range(this.state.range);
+    scale.domain(this.state.domain);
+    return scale;
+  }
+
+  getTicks() {
+    const scale = this.state.scale;
+    if (this.props.tickValues) {
+      return this.props.tickValues;
+    } else if (_.isFunction(scale.ticks)) {
+      return scale.ticks(this.props.tickCount);
+    } else {
+      return scale.domain();
+    }
   }
 
-  getYTransform() {
-    const range = this.state.xRange;
-    return "translate(" + range.min + "," + 0 + ")";
+  getTickFormat() {
+    const scale = this.state.scale;
+    if (this.props.tickFormat) {
+      return this.props.tickFormat;
+    } else if (_.isFunction(scale.tickFormat)) {
+      return scale.tickFormat(this.state.ticks.length);
+    } else {
+      return (x) => x;
+    }
   }
 
-  getXScale() {
-    const scale = this.props.scale().range([this.state.xRange.min, this.state.xRange.max]);
-    return scale.domain([this.props.xDomain.min, this.props.xDomain.max]);
+  getAxisPath() {
+    const orientation = this.props.orientation;
+    const range = this.state.range;
+    const sign = (orientation === "top" || orientation === "left") ? -1 : 1;
+    if (orientation === "top" || orientation === "bottom") {
+      return "M" + _.min(range) + "," + sign * this.props.outerTickSize +
+        "V0H" + _.max(range) + "V" + sign * this.props.outerTickSize;
+    } else {
+      return "M" + sign * this.props.outerTickSize + "," + _.min(range) + "H0V" +
+        _.max(range) + "H" + sign * this.props.outerTickSize;
+    }
   }
 
-  getYScale() {
-    const scale = this.props.scale().range([this.state.yRange.max, this.state.yRange.min]);
-    return scale.domain([this.props.yDomain.min, this.props.yDomain.max]);
+  getActiveScale() {
+    const scale = this.state.scale;
+    if (scale.rangeBand) {
+      return (x) => scale(x) + scale.rangeBand() / 2;
+    } else {
```

Comment:
no need to wrap the second `return` in an `else` since there's a return in the `if`already